gradle/lib/dependabot/gradle/file_fetcher.rb

```diff
@@ -14,6 +14,9 @@ class FileFetcher < Dependabot::FileFetchers::Base
       SUPPORTED_SETTINGS_FILE_NAMES =
         %w(settings.gradle settings.gradle.kts).freeze
 
+      # For now Gradle only supports libray .toml files in the main gradle folder
+      SUPPORTED_VERSION_CATALOG_FILE_PATH = %w(/gradle/libs.versions.toml).freeze
```

Comment:
That isn't always true. Using [`from` in the settings dsl](https://docs.gradle.org/current/javadoc/org/gradle/api/initialization/dsl/VersionCatalogBuilder.html#from-java.lang.Object-) you can import a toml catalog from any arbitrary path. `gradle/libs.versions.toml` is just the default convention that gradle will pickup without having to explicitly list it in your `settings.gradle[.kts]`

https://docs.gradle.org/current/userguide/platforms.html#sec:importing-catalog-from-file

I would suspect handling any toml file in the given [config directory](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#directory) would be safe? Or rather/maybe `root_dir/gradle/*.toml` + `root_dir/*.toml`? or maybe grep out lines using `from("*.toml")` to build the list of catalogs?

FWIW gradle does enforce that `from` can only be called once per catalog.
```kotlin
dependencyResolutionManagement {
    versionCatalogs {
        create("libs") {
            from("boom.versions.toml")
        }
    }
}
```
produces:
```text
Invalid catalog definition:
  - Problem: In version catalog libs, you can only call the 'from' method a single time.

* Try:
> Run with --info or --debug option to get more log output.

* Exception is:
org.gradle.api.InvalidUserDataException: Invalid catalog definition:
  - Problem: In version catalog libs, you can only call the 'from' method a single time.
    
    Reason: The method was called more than once.
    
    Possible solution: Remove further usages of the method call.
    
    Please refer to https://docs.gradle.org/7.6/userguide/version_catalog_problems.html#too_many_import_invocation for more details about this problem.
	at org.gradle.api.internal.catalog.problems.DefaultCatalogProblemBuilder.maybeThrowError(DefaultCatalogProblemBuilder.java:66)
[...]
```

Reply:
Sorry about the confusion created by the comment here! Your explanation and concern are valid, and your proposal looks good too.
The initial goal for this PR was to keep things simple and only support the default location of the `.toml` file, not considering any `dependencyResolutionManagement` extra configuration, leaving that task as a follow-up piece of work.

Reply:
So it wouldn't support declaring the version directly in the `settings.gradle` file then?

eg:
```
dependencyResolutionManagement {
    versionCatalogs {
        create("libs") {
            version("lombok", "1.18.24")
            library("lombok", "org.projectlombok", "lombok").versionRef("lombok")
        }
    }
}
```

Reply:
That's correct.

Reply:
Reticketed as https://github.com/dependabot/dependabot-core/issues/6831.

Reply:
Also reticketed my request as #6838